src/pages/ProductPage/ProductPage.tsx

```diff
@@ -27,7 +27,9 @@ const ProductPage = () => {
   const { cart, addProduct, decreaseAmount, increaseAmount } = useCartStore();
 
   const [initialAmount, setInitialAmount] = useState(1);
-  const productInCart = cart.find((product) => product.id === id);
+  const productInCart = cart.find(
+    (product: FavoriteProduct) => product.id === id
```

Comment:
Typescript get the type automatically here.
```suggestion
    (product) => product.id === id
```